include/solvers/vof.h

```diff
@@ -145,6 +141,7 @@ class VolumeOfFluid : public AuxiliaryPhysics<dim, GlobalVectorType>
       this->simulation_parameters,
       this->pcout,
       this->triangulation,
+      this->simulation_control->get_iteration_number(),
```

Comment:
Give the whole simulation control instead. It's super bad to move raw pointers around. This is verbotten.